### PR TITLE
[hls] Don't try to skip a stream if the offset is 0

### PR DESCRIPTION
**Pull request by back-to**
and don't raise KeyKerror if the m3u8 file is empty this allows the file to reload.

---

Fixed streamlink/streamlink#1419

streamlink/streamlink#1423 required for youtube.

### CODE REVIEW

File: src/streamlink/stream/hls.py [hunk 1]
```diff
@@ -136,15 +136,17 @@ def __init__(self, *args, **kwargs):
...
-        self.playlist_sequence = self.duration_to_sequence(self.duration_offset_start, self.playlist_sequences)
-        if self.duration_limit and self.playlist_end:
-            self.playlist_end = self.duration_to_sequence(self.duration_offset_start + self.duration_limit,
-                                                          self.playlist_sequences)
-        self.logger.debug("First Sequence: {0}; Last Sequence: {1}",
-                          self.playlist_sequences[0].num, self.playlist_sequences[-1].num)
-        self.logger.debug("Start offset: {0}; Duration: {1}; Start Sequence: {2}; End Sequence: {3}",
-                          self.duration_offset_start, self.duration_limit, self.playlist_sequence,
-                          self.playlist_end)
+        if self.duration_offset_start is not 0:
```
Comment (beardypig) [inline src/streamlink/stream/hls.py]: While `0` is a singleton in CPython, I think that `!=` should be used in this case :)

**Comment by beardypig** [comment]
I was just working on both these things ;)

File: src/streamlink/stream/hls.py (hunk 1 above)
Comment (back-to) [reply src/streamlink/stream/hls.py]: i changed it 👍

**Comment by gravyboat** [comment]
Thanks @back-to and thanks for reviewing @beardypig.

**Pull request merged**